jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/DefaultRule.java

```diff
@@ -243,11 +246,35 @@ private JExpression getDefaultSet(JType fieldType, JsonNode node) {
 
     }
 
-    private JExpression getDefaultEnum(JType fieldType, JsonNode node) {
+    private JType getParameterType(JDefinedClass fieldType, String methodName){
 
-        JInvocation invokeFromValue = ((JClass) fieldType).staticInvoke("fromValue");
-        invokeFromValue.arg(node.asText());
+        for (JMethod method : fieldType.methods()) {
+            if (EnumRule.FROM_VALUE_METHOD_NAME.equals(method.name())
+                    && (method.mods().getValue() & JMod.PUBLIC) == JMod.PUBLIC
+                    && (method.mods().getValue() & JMod.STATIC) == JMod.STATIC) {
+                final JType[] type = method.listParamTypes();
+                if (type.length != 1)
+                    throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' should has only one parameter in " + fieldType);
+                return type[0];
+            }
+        }
+        throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' is not found in " + fieldType);
+
+    }
 
+    private JExpression getDefaultEnum(JDefinedClass fieldType, JsonNode node) {
+
+        JInvocation invokeFromValue = fieldType.staticInvoke(EnumRule.FROM_VALUE_METHOD_NAME);
+        JType paramType = getParameterType(fieldType, EnumRule.FROM_VALUE_METHOD_NAME).unboxify();
+        if (paramType == paramType.owner().INT) {
```

Comment:
I prefere not to write spaghetti-code with dozens of if-else at all.
But you proposed me to move some more if-else from `getDefaultEnum` to 'getDefaultValue' and delete the `getDefaultEnum` method after that, is not it?
Well, not bad idea to remove the `getDefaultEnum` but I'll better use some AbstractFactory pattern right from the top `apply` method for this instead of if-else.